File: Tipo/C++/Concesionario.cc

# include "Auto.cc"
# include "Bus.cc"
# include "Moto.cc"
# include <iostream>

using namespace std;

int main() {

    // Vehiculo concesionario[3];

    // Con punteros se comporta raro ...

    Moto* m = new Moto("0x01", "Nexus", "110A", "4 Tiempos");
    Auto* a = new Auto("0x03", "Chery Tiggo", "8 Pro", 6);
    Bus* b = new Bus("0x02", "Yutong", "ZK6860", 3.50);

    // Usando metodos polimorficos

    m->mostrarDatos();
    m->tocarClaxon();
    cout << endl;

    a->mostrarDatos();
    a->tocarClaxon();
    cout << endl;

    b->mostrarDatos();
    b->tocarClaxon();
    cout << endl;

};

File: Tipo/Java/Concesionario.java

package tipos;

public class Concesionario {
	
	public static void main(String[] args) {
		
		// Vehiculo[] concesionario = new Vehiculo[3];
		Moto m = new Moto("0x01", "Nexus", "110A", "4 Tiempos");
		Bus b = new Bus("0x02", "Yutong", "ZK6860", 3.50);
		Auto a = new Auto("0x03", "Chery Tiggo", "8 Pro", 6);
		
		// Para el Objeto Moto
		m.mostrarDatos();
		m.tocarClaxon();
		System.out.println();
		
		// Para el Objeto Bus
		b.mostrarDatos();
		b.tocarClaxon();
		System.out.println();
		
		// Para el Objeto Auto
		a.mostrarDatos();
		a.tocarClaxon();
		System.out.println();
		
	}
	
}
